File: src/main/java/com/m2t/hachikodictionary/dto/word/WordDto.java
package com.m2t.hachikodictionary.dto.word;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class WordDto {
    private String id;
    private String title;
    private String kind;
    private String difficulty;
    private Set<String> description;
    private Set<String> synonyms;
    private Set<String> antonyms;
    private Set<String> sentences;
    private String fileUrl;
    private LocalDateTime audioFileCreatedAt;
    private String etymology;
}

File: src/main/java/com/m2t/hachikodictionary/dto/word/WordDto.kt
package com.m2t.hachikodictionary.dto.word

import java.time.LocalDateTime

data class WordDto(
    val id: String?,
    val title: String,
    val kind: String,
    val difficulty: String,
    val descriptions: Set<String>,
    val synonyms: Set<String>,
    val antonyms: Set<String>,
    val sentences: Set<String>,
    val fileUrl: String?,
    val audioFileCreatedAt: LocalDateTime?
) {
}

